tictactoe/adapter-implementation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/adapter/implementation/query/SampleAnswer.kt

```diff
@@ -0,0 +1,16 @@
+package com.anaplan.engineering.azuki.tictactoe.adapter.implementation.query
+
+import com.anaplan.engineering.azuki.core.system.Answer
+import com.anaplan.engineering.azuki.core.system.Check
+import com.anaplan.engineering.azuki.core.system.Query
+import com.anaplan.engineering.azuki.tictactoe.adapter.api.TicTacToeCheckFactory
+
+open class SampleAnswer<T> (
+    override val to: Query<T>,
+    override val value: T,
+    private val checkCreator: (TicTacToeCheckFactory) -> List<Check>
+) : Answer<T, TicTacToeCheckFactory> {
+
+    override fun createChecks(factory: TicTacToeCheckFactory) = checkCreator(factory)
+    override fun toString() = "Answer: $value"
+}
```

Comment:
I'm tempted to lift this out, given that it's extremely generic.  Something like


```suggestion
open class AnswerImpl<T, CF: CheckFactory> (
    override val to: Query<T>,
    override val value: T,
    private val checkCreator: (CF) -> List<Check>
) : Answer<T, CF> {

    override fun createChecks(factory: CF) = checkCreator(factory)
    override fun toString() = "Answer: $value"
}
```

Another possibility would be (I haven't checked if this compiles but hopefully you get the idea!):

```suggestion
fun <T, CF: CheckFactory> answer(
    to: Query<T>,
    value: T,
    checkCreator: (CF) -> List<Check> = listOf(UnsupportedCheck)
) : Answer<T, CF> = object : Answer<T, CF> {

    override val to = to
    override val value = value
    override fun createChecks(factory: CF) = checkCreator(factory)
    override fun toString = "Answer: $value"
}
```

Reply:
Probably worth doing, but it can wait for a second pass.